Darwin/Services/StitchRemoteMongoDBService/StitchSyncPerformanceTests/SyncL2ROnlyPerformanceTestDefinitions.swift

```diff
@@ -0,0 +1,251 @@
+        testHarness: SyncPerformanceIntTestHarness,
+        runId: ObjectId,
+        pctOfDocsWithChangeEvents: Double
+    ) {
+        let testName = "testL2R_SyncPass_${pctOfDocsWithChangeEvents}DocsChanged"
+        print(TAG, testName)
+
+        // Local variable for the number of docs updated in the test
+        // This should change for each iteration of the test.
+        var numberOfChangedDocs: Int?
+
+        testHarness.runPerformanceTestWithParameters(
+            testName: testName,
+            runId: runId,
+            testDefinition: { (ctx, numDocs, docSize) in
+                // Do the sync pass that will sync the
+                // local changes to the remote collection
+                guard let syncPassSucceeded = try? ctx.dataSynchronizer.doSyncPass(),
+                    syncPassSucceeded else {
+                    // Halt the test if the sync pass failed
+                    fatalError("sync pass failed")
+                }
+        }, beforeEach: { (ctx, numDocs, docSize) in
+            // Generate and insert the documents, and perform the initial sync.
+            print(TAG, "Setting up $testName test for $numDocs $docSize-byte docs")
+            var documentsForCurrentTest =
+                SyncPerformanceTestUtils.generateDocuments(numDoc: numDocs, docSize: docSize)
+
+            // Initial sync for a purely L2R scenario means inserting local documents,
+            // and performing a sync pass to sync those documents up to the remote.
+            let sync = ctx.coll.sync
+
+            // If sync fails for any reason, halt the test
+            sync.configure(
+                conflictHandler: DefaultConflictHandler<Document>.remoteWins(),
+                changeEventDelegate: nil,
+                errorListener: { error, id in
+                    print(
+                        "unexpected sync error with id " +
+                        "\(String(describing: id)): \(error)"
+                    )
+                    fatalError(error.localizedDescription)
+            })
+            sync.insertMany(&documentsForCurrentTest)
+
+            guard let syncPassSucceeded = try? ctx.dataSynchronizer.doSyncPass(),
+                syncPassSucceeded else {
+                // Halt the test if the sync pass failed
+                fatalError("sync pass failed")
+            }
+
+            // Randomly sample a percentage of the documents
+            // that will be locally updated
+            let shuffledDocs = documentsForCurrentTest.shuffled()
+
+            let docsToUpdate = pctOfDocsWithChangeEvents > 0.0 ?
+                shuffledDocs.prefix(upTo: Int(pctOfDocsWithChangeEvents * Double(numDocs))) :
+                []
+
+            docsToUpdate.forEach {
+                _ = sync.updateOne(
+                    filter: ["_id": $0["_id"] ?? BSONNull()],
+                    update: ["$set": ["newField": "blah"] as Document]
+                )
+            }
```

Comment:
This can be an updateMany() with $in. Also we should check that the result matchedCount and modifiedCount are the correct number